orderbook/src/main.rs

```diff
@@ -284,7 +285,7 @@ async fn main() {
         gas_price_estimator,
         native_token.address(),
         database.clone(),
-        args.shared.fee_discount_factor,
+        args.shared.fee_subsidy_factor,
```

Comment:
I would have slightly preferred to compute `fee_discoun_factor = 1.0 - options.fee_subsidy_factor` and passed that around everywhere as:
- It would require less code changes
- The computation could have been validated instead of having to remember to call validate each location that the arguments are parsed (in both `orderbook/src/main.rs` and `solver/src/main.rs`).
- Would have kept math simpler in various locations (no `1 - x` everywhere).

Reply:
That's cool too, I could change it to this in a follow up.

Reply:
I don't think its worth changing again. Just didn't get a chance to review last week and wanted to add my comments now.